File: LOOP_PROGRAM/Factorial.java
package LOOP_PROGRAM;

public class Factorial {
    /*
    * integer i,num,fact
     read num;
     set fact=1;
     * for i=num to 1 step -1
     * fact=fact*i
     * end for
     * print fact
     * set i=num
     * while(i>=1)
     * fact=fact*i
        i--
* end while      */
    public static void main(String[] args) {
        int num=5;
        int fact=1;
        for(int i=num;i>=1;i--){
            fact=fact*i;
        }
        System.out.println(fact);
    }
}

File: Factorial.cpp
// FUNCTION FACTORIAL(N:INTEGER):INTEGER
//   DECLARE FACT:INTEGER
//   SET FACT=1
//   FOR I IN N TO 1 STEP -1
//      FACT = FACT * I
//   END FOR
//   RETURN FACT
// END FUNCTION

#include <iostream>
using namespace std;

long factor(int n) {
    long fact = 1;
    for (int i = n; i > 1; i--) {
        fact *= i;
    }
    return fact;
}

int main() {
    int x;
    cin >> x;
    cout << factor(x) << endl;
    return 0;
}
